Add ros_bt_py technology info + examples

Oberacda opened the pull request.

Oberacda commented:
@sea-bass As you are the only one I know that tried `ros_bt_py`, do you think the README is sufficient for people who have not worked with it before? Should I put the examples from the presentation in the readme as well?

Oberacda commented:
Otherwise, this is finished on my end.

sea-bass commented:
I'm also still running somewhat frequently into the PyRoboSim display sync issue. I will take a look to see if I can improve this...

sea-bass commented:
> ### Problem 4
> I think you're navigating the robot to the `kitchen` room instead of one of the actual locations (fridge/pantry), so my robot got stuck trying to place the `bread0` object over and over again because it was not in the right location. Please take a look at that.

I think this specifically happens when the robot is holding the `bread0` or `butter0` object and, on the way to place it on the table, it has to go to the charger due to low battery. There may be a subtle behavior bug there.

Oberacda commented:
> > ### Problem 4
> > I think you're navigating the robot to the `kitchen` room instead of one of the actual locations (fridge/pantry), so my robot got stuck trying to place the `bread0` object over and over again because it was not in the right location. Please take a look at that.
> 
> I think this specifically happens when the robot is holding the `bread0` or `butter0` object and, on the way to place it on the table, it has to go to the charger due to low battery. There may be a subtle behavior bug there.

I briefly looked into this just now, and I think it is an issue with the interrupt during the execution of the path following. As far as I can see it flags the interrupted movement as successful and just continues with the placing after returning from the charger.

I will follow up on this tomorrow morning.

Oberacda commented on technologies/ros_bt_py/README.md:
```diff
@@ -0,0 +1,89 @@
+>
+> BUG: When updating node attributes and clicking apply, a popup will say that changes are discarded. This a bug and the changes will apply fully as intended.
+> This will be fixed with the next GUI update.
+>
```
No unfortunately not, these were all backend fixes. We are currently writing a new UI for the library from scratch, as the old one has much technical debt from the ROS1 version. Thus, we have not made any fixes to the current UI in a while. I was hoping the new one would be ready for ROSCon, but its still too unstable.

sea-bass commented:
> I briefly looked into this just now, and I think it is an issue with the interrupt during the execution of the path following. As far as I can see it flags the interrupted movement as successful and just continues with the placing after returning from the charger.
> 
> I will follow up on this tomorrow morning.

Are you officially canceling the original ROS action, or just sending a new one? If it's the second, then I could see how this happens.

Oberacda commented on technologies/ros_bt_py/ros_bt_py_pyrobosim/trees/close_door.yaml:
```diff
@@ -0,0 +1,215 @@
+      - CloseDoorTask
+      - CloseDoorAction
+      - GetCloseDoorResult
+      - CloseDoorSuccessfull?
```
Thanks. Question marks in Node names are allowed, so this should be fine. I also fixed the typo.

Oberacda commented:
> The examples all ran for me -- just a few things.
> 
> ## README / Instructions
> I think we need some instructions on how to load trees from the `ros_bt_py_pyrobosim` package. It took me a while to figure out.

I added a section about loading the example trees. Hopefully, this is more clear now.

> 
> Also, generally the UX of this is interesting. It seems like if I select the wrong tree by accident by single-clicking, I have to hit "Back"? Also I get no visual confirmation on which tree I have actually selected.
> 

I am aware that the UX needs some polishing. I will try to find some time to look into it,

> ## Problems
> ### Problem 1
> Worked well!
> 
> ### Problem 2
> Worked well. Suggest renaming `PlanToDump` (and similar) to say `Dumpster` instead of `Dump`, as for English speakers this could... be funny naming.
> 

I have renamed the nodes.

> ### Problem 3
> Worked well. Question, is there a way to look inside the subtrees here while the trees are running? I couldn't figure out how to expand them.

I added a section about this to the README. You need to check the PublishSubtree box in the topbar and then you can inspect them via the Tree Drop Down above the editor.
> 
> ### Problem 4
> I think you're navigating the robot to the `kitchen` room instead of one of the actual locations (fridge/pantry), so my robot got stuck trying to place the `bread0` object over and over again because it was not in the right location. Please take a look at that.
> 
> I tried again and it seems it ran fine, but only partially. The bread and butter was brought to the table, but nothing was done with the waste/dumpster part of the objective. Maybe that is not implemented yet in this solution?

I indeed forgot to implement the dumpster part in the previous version. I added it alongside a fix for the issues regarding the recharging being triggered while holding an object.

sea-bass reviewed (approved):
Thanks! Let's merge it.

sea-bass commented:
... after the lint fixes are resolved.
